### PR TITLE
Add updated config parameters to UPGRADE-3.0.md

### LINKED ISSUE #8831: Updated configuration parameters in UPGRADE-3.0.md are incomplete
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
While working on the M2 > M3 upgrade script, I came across some settings in `local.php` that have changed in Mautic 3, but are not mentioned at https://github.com/mautic/mautic/blob/3.x/UPGRADE-3.0.md#configuration. This is leading to problems (**e.g. old update URL (not the new GitHub one)**), so I decided to compare a full `local.php` file (generated when going to Configuration > Save & Close in the UI) between the two versions.

**It is absolutely crucial that we properly migrate settings to their new versions, as otherwise we'll very likely break existing Mautic installations post-upgrade.**

Please find below all the parameters that differ between **clean installations** of Mautic 2 and Mautic 3. Please note the "Comment" column in the table as well (vertically scroll to the right)!!

**I'll need to have a decision made on every single config parameter below in order to prevent issues when users upgrade their Mautic instances from 2 to 3.**

Google Sheets version of the table below (only editable by Mautic's Product team for security reasons): https://docs.google.com/spreadsheets/d/1Fr_VBuiDbYy_YdjPMCFsFIyUP5vz6nexiNg9BNarluM/edit?usp=sharing 

| Mautic 2 local.php | Mautic 3 local.php | Comment
|---|---|---|
| 'mailer_transport' => 'mail' | 'mailer_transport' => 'sendmail'  | The `mail` option that was present in v2 now leads to `Not a built-in Swiftmailer transport: mail` errors in v3. Not sure if there are any other values that were present in v2 but not in v3 anymore! (TODO check)
| 'mailer_spool_path' => '%kernel.root_dir%/spool' | 'mailer_spool_path' => '%kernel.root_dir%/../var/spool' | Should I only replace this value when the M2 value exactly matches what is shown on the left, or should I replace `%kernel.root_dir%` with `%kernel.root_dir%/../var`?
| 'cache_path' => '/var/www/html/app/cache' | 'cache_path' => '/var/www/html/app/../var/cache' | Probably the smartest to replace `app/cache` with `app/../var/cache` so that we don't mess with this path if folks have set a custom cache path?
| 'log_path' => '/var/www/html/app/logs' | 'log_path' => '/var/www/html/app/../var/logs' | Probably the smartest to replace `app/logs` with `app/../var/logs` so that we don't mess with this path if folks have set a custom logs path?
| 'tmp_path' => '/var/www/html/app/cache' | 'tmp_path' => '/var/www/html/app/../var/tmp' | (Note this was written to the `cache` folder previously). Probably the smartest to replace `app/cache` with `app/../var/tmp` so that we don't mess with this path if folks have set a custom tmp path?
| 'dev_hosts' => null | 'dev_hosts' => array() | Replace null with array
| 'rememberme_lifetime' => '31536000' | 'rememberme_lifetime' => 31536000 | Replace string with int? Or leave as-is?
| N/A | 'cookie_httponly' => false | add `cookie_httponly` parameter? Or leave as-is (non-existent)?
| N/A | 'track_private_ip_ranges' => false | add `track_private_ip_ranges` parameter? Or leave as-is (non-existent)?
| N/A | 'batch_campaign_sleep_time' => false | add `batch_campaign_sleep_time` parameter? Or leave as-is (non-existent)?
| 'system_update_url' => 'https://updates.mautic.org/index.php?option=com_mauticdownload&task=checkUpdates' | 'system_update_url' => 'https://api.github.com/repos/mautic/mautic/releases' | Replace update URL with new one
| 'theme' => 'Mauve' | 'theme' => 'blank' | Only replace theme if it equals Mauve (otherwise we'd potentially reset the theme the user has selected)
| N/A | 'debug' => false | add `debug` parameter? Or leave as-is (non-existent)?
| N/A | 'stats_update_url' => 'https://updates.mautic.org/stats/send' | add `stats_update_url` parameter? Or leave as-is (non-existent)?
| N/A | 'install_source' => 'Mautic' | add `install_source` parameter? Or leave as-is (non-existent)?
| 'api_oauth2_access_token_lifetime' => 60 | 'api_oauth2_access_token_lifetime' => 3600 | update to 3600 if current value equals 60?
| 'api_oauth2_refresh_token_lifetime' => 14 | 'api_oauth2_refresh_token_lifetime' => 1209600 | update to 1209600 if current value equals 14?
| 'api_rate_limiter_cache'  => [ 'type' => 'filesystem'] | 'api_rate_limiter_cache'  => [ 'adapter' => 'cache.adapter.filesystem'] | Already in UPGRADE-3.0.md. But, do we only account for the `filesystem` case? Or do we need to migrate this config in other cases as well (like with memcached, [as mentioned in the Mautic docs](https://developer.mautic.org/#api-rate-limiter))?
| 'email_frequency_number' => null | 'email_frequency_number' => 0 | if `null` then update to `0` or leave as-is?
| 'email_frequency_time' => null | 'email_frequency_time' => 'DAY' | if `null` then update to `DAY` or leave as-is?
| 'track_by_fingerprint' => 0 | N/A | Remove or leave as-is?
| 'rabbitmq_host' => 'localhost', 'rabbitmq_port' => '5672', 'rabbitmq_vhost' => '/', 'rabbitmq_user' => 'guest', 'rabbitmq_password' => 'guest', 'beanstalkd_host' => 'localhost', 'beanstalkd_port' => '11300', 'beanstalkd_timeout' => '60' | N/A | Remove or leave as-is?
| 'report_export_batch_size' => '1000' | 'report_export_batch_size' => 1000 | convert string to int or leave as-is?
| 'report_export_max_filesize_in_bytes' => '5000000' | 'report_export_max_filesize_in_bytes' => 5000000 | convert string to int or leave as-is?
| N/A | 'sms_enabled' => false | add `sms_enabled` parameter? Or leave as-is (non-existent)?
| 'sms_frequency_number' => null | 'sms_frequency_number' => 0 | Convert `null` to `0` or leave as-is?
| 'sms_frequency_time' => null | 'sms_frequency_time' => 'DAY' | convert `null` to `DAY` or leave as-is?
| 'saml_idp_own_password' => null | N/A | Remove or leave as-is?
| 'webhook_start' => '0' | N/A | Remove or leave as-is?
| 'webhook_limit' => '10', 'webhook_log_max' => '1000', 'webhook_disable_limit' => '100', 'webhook_timeout' => '15' | 'webhook_limit' => 10, 'webhook_log_max' => 1000, 'webhook_disable_limit' => 100, 'webhook_timeout' => 15 | Convert string to int or leave as-is?

| Q   | A
| --- | ---
| Mautic version | 3.0.0-beta2
| PHP version | 7.2
| Browser | N/A

### Steps to reproduce
1. Download & install a fresh Mautic 2 instance.
2. Go through the installation process, click Configuration > Save & Close. This will generate a full-blown `app/config/local.php` file with many parameters in it.
3. Repeat step 1 and 2 for Mautic 3 beta2
4. Compare both `local.php` files (attached 2 example files to this issue for convenience, you can show them side by side in e.g. a text editor): [local_m2_and_m3_files.zip](https://github.com/mautic/mautic/files/4678805/local_m2_and_m3_files.zip)
 
### Log errors
N/A (multiple, e.g. when trying `mautic:update:find` while the old update URL is still in place)

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

### PR DESCRIPTION
**Pull request by dennisameling**
Closes #8831, adds updated config parameters to UPGRADE-3.0.md. Only textual changes

### CODE REVIEW

File: UPGRADE-3.0.md [hunk 1]
```diff
@@ -31,12 +31,45 @@
...
 ### Configuration
 
-If you have `api_rate_limiter_cache` present in you `app/config/local.php`, you need to change the value to:
-
+In Mautic 3, several configuration parameters have changed (`app/config/local.php`). The [upgrade script](https://github.com/mautic/mautic/issues/8819) will update those automatically for you, but here's an overview of parameters that will be updated by the script:
+
+| Key | Old value | New value | Comment
+|---|---|---|---|
+| mailer_transport | 'mail' | 'sendmail' | 'mail' option was removed in SwiftMailer 6, other options should keep working as-is
+| system_update_url | 'https://updates.mautic.org/ index.php?option=com_mauticdownload& task=checkUpdates' | 'https://api.github.com/ repos/mautic/mautic/releases' | new update mechanism
```
Comment (RCheesley) [inline UPGRADE-3.0.md]:
```suggestion
| system_update_url | 'https://updates.mautic.org/ index.php?option=com_mauticdownload& task=checkUpdates' | 'https://api.github.com/ repos/mautic/mautic/releases' | New update mechanism
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Just one tiny grammar suggestion!

**Review by npracht — APPROVED** [review]
Wow this is very cool work for community people 👏 
Beside Ruth feedback, i don't see any issue.

**Review by RCheesley — APPROVED** [review]
:+1:

**Pull request merged**